backend/Controllers/RecipeController.js

```diff
@@ -114,80 +115,158 @@ const allRecipe = async (req, res) => {
  * @description Uploads Image to a github repo and returns the downloadable link
  * @access private
  */
+// const imageToGithub = async (fileImage, name, unique) => {
+//   const owner = process.env.OWNER; 
+//   const repo =process.env.REPO; 
+//   const branch =process.env.BRANCH; 
+// console.log(owner,repo,branch)
+//   const base64Content = fileImage.split(';base64,').pop(); // Extract base64 content
+//   const fileContent = Buffer.from(base64Content, 'base64').toString('base64'); // Re-encode to base64
+//   const path = `TastyTrails/Recipe/${unique}${name}`; 
+//   const message = `Add ${unique} ${name} via API`; // Commit message
+//   const url = `https://api.github.com/repos/${owner}/${repo}/contents/${path}`;
+// console.log(url)
+//   try {
+//     // Step 1: Check if the file already exists
+//     let fileExists = false;
+   
+//     try {
+//       const response = await axios.get(url, {
+//         headers: {
+//           Authorization: `TOKEN ${process.env.TOKEN}`,
+//           'X-GitHub-Api-Version': '2022-11-28',
+//         },
+//       });
+//       console.log(response)
+//       if (response.status === 200) {
+//         fileExists = true;
+//       }
+//     } catch (error) {
+//       // If file does not exist (404), proceed with creating it
+//       if (error.response && error.response.status === 404) {
+//         fileExists = false;
+//       } else {
+//         throw error;
+//       }
+//     }
+
+//     // Step 2: Upload or update the file based on existence
+//     const requestPayload = {
+//       message,
+//       content: fileContent,
+//       branch,
+//     };
+
+//     if (fileExists) {
+//       // If file exists, get the sha and update it
+//       const sha = (await axios.get(url, {
+//         headers: {
+//           Authorization: `TOKEN ${process.env.TOKEN}`,
+//           'X-GitHub-Api-Version': '2022-11-28',
+//         },
+//       })).data.sha;
+
+//       requestPayload.sha = sha; // Required for updates
+//     }
+
+//     // Step 3: Upload or update the file
+//     const response = await axios.put(url, requestPayload, {
+//       headers: {
+//         Authorization: `TOKEN ${process.env.TOKEN}`,
+//         'X-GitHub-Api-Version': '2022-11-28',
+//       },
+//     });
+
+//     if (response.status === 201 || response.status === 200) {
+//       return response.data.content.download_url; // Return the URL of the uploaded image
+//     } else {
+//       console.error('Failed to upload image:', response.status, response.statusText);
+//       return null;
+//     }
+
+//   } catch (error) {
+//     console.error('Error uploading image:', error.response ? error.response.data : error.message);
+//     return null;
+//   }
+// };
+
 const imageToGithub = async (fileImage, name, unique) => {
```

Comment:
@Saurabhchaudhary9799 why did you comment out the above code? Try to incorporate your code in the existing code only. Or if this part is not needed please remove it do not leave on comment & make sure your changes does not break any other feature

Reply:
@AlfiyaSiddique , when i am doing the setup of this project on local machine ,  imageToGIthub function code is not working , then i updated the code and commented first written code . I want to work the comment section issue that are done on recipe , so i have to create an recipe and recipe creation is going to failed again and again because of imageToGithub function.